File: Leaf-Similar Trees/leaf-similar-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        vector<int>v1;
        vector<int>v2;
        calcLeaf(root1,v1);
        calcLeaf(root2,v2);
        return v1==v2;
    }
    void calcLeaf(TreeNode* r,vector<int> &v){
        if(r==NULL)
        return;
        if(r->left==NULL && r->right==NULL)
        v.push_back(r->val);
        calcLeaf(r->left,v);
        calcLeaf(r->right,v);
    }
};

File: Leaf-Similar Trees/leaf-similar-trees.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def leafSimilar(self, root1, root2):
        """
        :type root1: Optional[TreeNode]
        :type root2: Optional[TreeNode]
        :rtype: bool
        """
        def collectLeaves(node):
            if not node:
                return []
            if not node.left and not node.right:
                return [node.val]
            return collectLeaves(node.left) + collectLeaves(node.right)
        
        return collectLeaves(root1) == collectLeaves(root2)
